gotrue/types.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+
+from enum import Enum
+from time import time
+from typing import Any, Callable, Dict, List, Optional, Type, TypeVar, Union
+
+from httpx import Response
+from pydantic import BaseModel, root_validator
+
+from gotrue.helpers import check_response
+
+T = TypeVar("T", bound=BaseModel)
+
+
+def determine_session_or_user_model_from_response(
+    response: Response,
+) -> Union[Type[Session], Type[User]]:
+    if "access_token" in response.json():
+        return Session
+    return User
+
+
+class BaseModelFromResponse(BaseModel):
+    @classmethod
+    def parse_response(cls: Type[T], response: Response) -> T:
+        check_response(response)
+        return cls.parse_obj(response.json())
+
+
+class CookieOptions(BaseModelFromResponse):
+    name: str
+    """The name of the cookie. Defaults to `sb:token`."""
+    lifetime: int
+    """The cookie lifetime (expiration) in seconds. Set to 8 hours by default."""
+    domain: str
+    """The cookie domain this should run on.
+    Leave it blank to restrict it to your domain."""
+    path: str
+    same_site: str
+    """SameSite configuration for the session cookie.
+    Defaults to 'lax', but can be changed to 'strict' or 'none'.
+    Set it to false if you want to disable the SameSite setting."""
+
+
+class Identity(BaseModelFromResponse):
+    id: str
+    user_id: str
+    provider: str
+    created_at: str
```

Comment:
Shouldn't these kind of fields be annotated as `datetime.datetime` for pydantic to parse them into datetime objects?

Reply:
Hm, I only mean the fields like `created_at`. Misclicked and selected more lines I think 🤔

Reply:
I declared it as `str` for parity with gotrue-js types, do we declare them as `datetime` even if it breaks parity?

Reply:
I think it would be a good idea to have them as datetime objects, will be quite convenient for end users IMO

Reply:
What do you think @dreinon ?

Reply:
I would rather use datetime as @anand2312 said. For me, it's more convenient even if breaks parity, le's just make sure we mention thisbin the docs (or README for now).
Nevertheless, we could ask js developers why they didn't cast date strings to DateTime types there and see if they convince us haha.

Reply:
I add `datetime` and also `UUID` for ids

Reply:
Yes, README should be generally updated